Libs/GdkPixbuf-2.0/Classes/Pixbuf.cs

```diff
@@ -14,8 +14,11 @@ public partial class Pixbuf
 
         public static Pixbuf NewFromFile(string fileName)
         {
-            IntPtr handle = Native.Pixbuf.Instance.Methods.NewFromFile(fileName, out var error);
+            var error = new GLib.Native.Error.Handle(IntPtr.Zero);
```

Comment:
Why do we need to define the error beforehand? The SafeHandle should free the memory correctly?

I did some tests and it looks like the error handle is not freed (immediately) if the `GException` saves the error safe handle. But if we do not hold on to it, the reference it is freed immediately. I wonder if the garbage collector would collect the safe handle and thus free the error at a later point in time?

If possible I want to keep the original code. The big point for safe handle is that it removes the need to care about the unmanaged memory freeing (e.g. `error.Dispose()`).

Reply:
We had quite a lot of trouble with `SafeHandle` + direction attributes when getting `TypeQuery` to work (`AccessViolationException`). The solution was to not generate direction attributes for SafeHandle-type parameters (i.e. records). The downside is this means that error handling can't use `out` anymore as GError is a record.

The calls to Dispose can be removed (in fact, they should happen automatically when the variable leaves the block IIRC).

Reply:
Did the problem regarding SafeHandles occur only with the error argument or with other arguments, too?

I reread the documentation regarding error handling: https://developer.gnome.org/glib/2.64/glib-Error-Reporting.html

I read it like it is correct to pass a pointer. But I assume that if we mark a safe handle as `out` the marshaller is implicitly creating a pointer because C does not know of `out` parameters and then returning the result into our managed world. This can be the reason why it is working. So it could be correct to do it in this way as in some cases it is more "idiomatic" like in the error case.

Parameters are described like this:

```xml
<parameter name="padding_top" direction="out" caller-allocates="0" transfer-ownership="full" optional="1" allow-none="1">
```

What happens if we use `out` parameters only if direction is `out` and `caller-allocates` is 0? In case `caller-allocates` is 1 we just generate a regular parameter.

Just for reference, the error parameter is generated here as `out` and `caller-allocates` = 0
https://github.com/gircore/gir.core/blob/50d797b6e01c1caec9304fd03833ace359626598/Repository/Factories/Model/ParameterListFactory.cs#L37

Can you verify this? If this works out I think our native code could be quite a bit cleaner.